returnn/__main__.py

```diff
@@ -231,6 +231,11 @@ def load_data(config, cache_byte_size, files_config_key, **kwargs):
         kwargs["cache_byte_size"] = cache_byte_size
     Dataset.kwargs_update_from_config(config, kwargs)
     data = init_dataset(kwargs)
+  elif config.is_typed(files_config_key) and isinstance(config.typed_value(files_config_key), Dataset):
+    # Config contains dataset object, load it
+    data = config.typed_value(files_config_key)
+    data.initialize()
```

Comment:
Using `init_dataset(data)` (from datasets.basic) would be a bit more consistent.

Reply:
But as said, I think we should not allow to have a `Dataset` instance directly here. It should be a `callable`.